### ISSUE TITLE
Hardcoded kube-proxy version in eksctl

### ISSUE THREAD

**Issue by greenu** [weaveworks/eksctl#3822]
**What were you trying to accomplish?**
Upgrade kube-proxy.

**What happened?**
kube-proxy is being downgraded `kube-proxy:v1.20.4-eksbuild.2` -> `kube-proxy:v1.20.4-eksbuild.1`
This is because of hardcoded value in kube_proxy.go:
https://github.com/weaveworks/eksctl/blob/8c022c9b8d124e9a2fdb7a09820e5a5ffec529d4/pkg/addons/default/kube_proxy.go#L159

**How to reproduce it?**
* create EKS 1.20 with eksctl. We are not using EKS addons yet.
* `eksctl utils update-kube-proxy --cluster ${CLUSTER_NAME} --region ${AWS_REGION} --approve`

**Logs**
```
$ kubectl describe daemonset kube-proxy --namespace kube-system | grep Image | cut -d "/" -f 3
kube-proxy:v1.20.4-eksbuild.2

2021-06-03 14:42:23 [ℹ]  eksctl version 0.51.0
2021-06-03 14:42:23 [ℹ]  using region us-west-2
2021-06-03 14:42:23 [ℹ]  "kube-proxy" is now up-to-date

<...wait until rollout completes...>

$ kubectl describe daemonset kube-proxy --namespace kube-system | grep Image | cut -d "/" -f 3
kube-proxy:v1.20.4-eksbuild.1
```

**Versions**
```
$ eksctl version
0.51.0 
$ kubectl version
1.20.4
```

**Comment by Callisto13** [comment]
Thanks for catching this @greenu. Feel free to submit a fix, otherwise the team will get to it when they can 👍

**Comment by pradeepnnv** [comment]
I'd like to work on this issue.

**Comment by Callisto13** [comment]
👍  all yours @pradeepnnv

**Comment by aclevername** [comment]
Closing this issue, see https://github.com/weaveworks/eksctl/pull/4091

TLDR: We don't have a good way of detecting what the latest `build.X` is for kube-proxy, at the moment we use your kubernetes version `x.y.z-build.1` by default, and this will always work. It might be that a `build.2` exists, or it might not. So for a best effort approach at upgrading we will just bump to `build.1`. If we can find a way of detecting higher builds then we can change approach, but atm I don't know a way of doing this.
